src/DynamoRevit/DynamoRevit.cs

```diff
@@ -551,16 +558,30 @@ private static void TryOpenAndExecuteWorkspaceInCommandData(DynamoRevitCommandDa
             if (commandData.JournalData.ContainsKey(JournalKeys.DynPathKey))
             {
                 bool isAutomationMode = CheckJournalForKey(commandData, JournalKeys.AutomationModeKey);
-                bool forceManualRun = CheckJournalForKey(commandData, JournalKeys.ForceManualRunKey);                
+                bool forceManualRun = CheckJournalForKey(commandData, JournalKeys.ForceManualRunKey);
 
-                if (ModelState == RevitDynamoModelState.StartedUIless)
+                bool useExistingWorkspace = false;
+                if (CheckJournalForKey(commandData, JournalKeys.DynPathCheckExisting))
                 {
-                    revitDynamoModel.OpenFileFromPath(commandData.JournalData[JournalKeys.DynPathKey], forceManualRun);
+                    WorkspaceModel currentWorkspace = revitDynamoModel.CurrentWorkspace;
+                    if (currentWorkspace.FileName.Equals(commandData.JournalData[JournalKeys.DynPathKey], 
+                        StringComparison.OrdinalIgnoreCase))
+                    {
+                        useExistingWorkspace = true;
+                    }
                 }
-                else
+
+                if (!useExistingWorkspace)
```

Comment:
can you add a quick comment to this block - 
something like,  "if use existing is false, open the specified workspace"